bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/Stream2JSONInputStream.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.rest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.IOUtils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * This {@link InputStream} will stream {@link Stream}s as JSON one item at a time. This will reduce memory usage when
+ * streaming large collections through the REST interface. The input stream creates one JSON representation at a time
+ * from the top level elements of the stream. For best performance a flattened stream should be provided. Otherwise a
+ * nested collections JSON representation will be fully transformed into memory.
+ *
+ * @author Henning Treu - initial contribution
+ *
+ */
+public class Stream2JSONInputStream extends InputStream {
+
+    private final Iterator<String> iterator;
+
+    private InputStream jsonElementStream;
```

Comment:
Shouldn't we implement the Closeable interface and ensure the input stream is closed if the close function is called (and ensure that the close function is called by the user of this class)?

Isn't there a resource leak if all the input stream are never closed?